pkg/virt-handler/vm_test.go

```diff
@@ -0,0 +1,107 @@
+package virthandler
+
+import (
+	"net/http"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/ghttp"
+	"k8s.io/client-go/pkg/api"
+	kubev1 "k8s.io/client-go/pkg/api/v1"
+	metav1 "k8s.io/client-go/pkg/apis/meta/v1"
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/runtime/serializer"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/logging"
+)
+
+var _ = Describe("PVC", func() {
+	RegisterFailHandler(Fail)
+	logging.DefaultLogger().SetIOWriter(GinkgoWriter)
+
+	expectedPVC := kubev1.PersistentVolumeClaim{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PersistentVolumeClaim",
+			APIVersion: "v1",
+		},
+		Spec: kubev1.PersistentVolumeClaimSpec{
+			VolumeName: "disk-01",
+		},
+		Status: kubev1.PersistentVolumeClaimStatus{
+			Phase: kubev1.ClaimBound,
+		},
+	}
+
+	source := kubev1.ISCSIVolumeSource{
+		IQN:          "iqn.2009-02.com.test:for.all",
+		Lun:          1,
+		TargetPortal: "127.0.0.1:6543",
+	}
+
+	expectedPV := kubev1.PersistentVolume{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PersistentVolume",
+			APIVersion: "v1",
+		},
+		Spec: kubev1.PersistentVolumeSpec{
+			PersistentVolumeSource: kubev1.PersistentVolumeSource{
+				ISCSI: &source,
+			},
+		},
+	}
+
+	server := ghttp.NewServer()
+	server.AppendHandlers(
+		ghttp.CombineHandlers(
+			ghttp.VerifyRequest("GET", "/apis/kubevirt.io/v1alpha1/namespaces/default/pvc/test-claim"),
```

Comment:
Since this is a core kubernetes client, you have to go to `/api/v1` or `apis/v1`. The core rest client will know where to look.